### PR TITLE
Monkey patch greenlet celery pools

### LINKED ISSUE #8023: CeleryExecutor gevent/eventlet pools need monkey patching
**Apache Airflow version**:
1.10.9

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T23:42:50Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.10-gke.17", GitCommit:"bdceba0734835c6cb1acbd1c447caf17d8613b44", GitTreeState:"clean", BuildDate:"2020-01-17T23:10:13Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"linux/amd64"}
```
Note: the issue is not specific to k8s.

**Environment**:
Any. I was able to reproduce using [CeleryExecutor docker-compose](https://github.com/puckel/docker-airflow/blob/1.10.9/docker-compose-CeleryExecutor.yml) in the puckel repo (code version tagged as 1.10.9).

**What happened**:
When setting `pool` in the `[celery]` section [in airflow.cfg](https://github.com/apache/airflow/blob/master/airflow/config_templates/default_airflow.cfg#L598) to `eventlet` or `gevent`, task instances get scheduled, queued, picked up by the workers, **but not executed**.

**What you expected to happen**:
Task instances should be executed. The problem is that the application is not [monkey-patched](https://eventlet.net/doc/patching.html#monkey-patch). Celery by default handles monkey-patching but not in all scenarios (e.g. only if Celery is invoked via command line, [more info](https://github.com/celery/celery/search?q=maybe_patch_concurrency&unscoped_q=maybe_patch_concurrency)).

Airflow invokes Celery workers in Python via [.run()](https://github.com/celery/celery/blob/v4.2.2/celery/bin/worker.py#L232). Unfortunately, this function does not handle monkey patching.

**How to reproduce it**:
1. Clone [puckel's docker-airflow](https://github.com/puckel/docker-airflow/tree/1.10.9):
    ```
    git clone git@github.com:puckel/docker-airflow.git
    ```
2. Modify Dockerfile to:
    ```
    RUN pip install eventlet
    ```
    Then:
    ```
    docker build --rm -t puckel/docker-airflow:1.10.9 .
    ```
3. Set `pool = eventlet` in [airflow.cfg](https://github.com/puckel/docker-airflow/blob/1.10.9/config/airflow.cfg#L509) (the file will be mounted by docker-compose).

4. Spin up [the CeleryExecutor docker compose]([CeleryExecutor docker-compose](https://github.com/puckel/docker-airflow/blob/1.10.9/docker-compose-CeleryExecutor.yml):
    ```
    docker-compose -f docker-compose-CeleryExecutor.yml up -d
    ```
5. Navigate to http://localhost:8080, and run an example DAG.
6. Notice that no task ever gets to the running state.

**Solution**:
Ideally this should be fixed in [Celery](git@github.com:puckel/docker-airflow.git), but in the meantime it might be good to have a solution here as well. Here is a patch that I applied to solve this (on Airflow 1.10.9):
```
--- cli.py	2020-03-27 17:05:45.000000000 -0400
+++ cli-new.py	2020-03-27 17:19:48.000000000 -0400
@@ -1098,7 +1098,10 @@
     }

     if conf.has_option("celery", "pool"):
-        options["pool"] = conf.get("celery", "pool")
+        pool = conf.get("celery", "pool")
+        options["pool"] = pool
+        from celery import maybe_patch_concurrency
+        maybe_patch_concurrency(['-P', pool])

     if args.daemon:
         pid, stdout, stderr, log_file = setup_locations("worker",
```

### PR DESCRIPTION
**Pull request by aamangeldi**
Celery pools of type eventlet and gevent use greenlets, which requires monkey patching the app.
[Here](https://eventlet.net/doc/patching.html#monkey-patch) is a description from Eventlet docs.

Otherwise task instances hang on the workers and are never executed.
Related issue: #8023 

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by aamangeldi** [comment]
Unit tests passed locally, I'm not quite sure how to upload test coverage to Codecov. Any help would be appreciated. Thanks!

File: airflow/cli/commands/celery_command.py [hunk 1]
```diff
@@ -125,7 +125,10 @@ def worker(args):
     }
 
     if conf.has_option("celery", "pool"):
-        options["pool"] = conf.get("celery", "pool")
+        pool = conf.get("celery", "pool")
+        options["pool"] = pool
+        from celery import maybe_patch_concurrency
```
Comment (turbaszek) [inline airflow/cli/commands/celery_command.py]: Let's move this import to the top, we are importing whole celery anyway

File: airflow/cli/commands/celery_command.py [hunk 2]
```diff
@@ -125,7 +125,10 @@ def worker(args):
...
+        maybe_patch_concurrency(['-P', pool])
```
Comment (turbaszek) [inline airflow/cli/commands/celery_command.py:136]: Should we leave here a short note of why we are doing this?

**Comment by turbaszek** [comment]
> Unit tests passed locally, but I'm not quite sure how to upload test coverage to Codecov. Any help would be appreciated. Thanks!

I think we currently disabled Codecov, but it's not a problem ;)

File: airflow/cli/commands/celery_command.py (hunk 2 above)
Comment (aamangeldi) [reply airflow/cli/commands/celery_command.py:136]: Good idea!

File: airflow/cli/commands/celery_command.py (hunk 1 above)
Comment (aamangeldi) [reply airflow/cli/commands/celery_command.py]: Sounds good.

**Review by turbaszek — APPROVED** [review]
Looks good to me 🚀

**Comment by turbaszek** [comment]
Thanks @aamangeldi ! 🚀

**Pull request merged**